docs/src/main/sphinx/security/user-mapping.rst

```diff
@@ -2,67 +2,123 @@
+username to emit after the regex substitution. If the regular expression does
+not match the incoming username, authentication is denied.
+
+Specify your regex pattern in the appropriate property in your coordinator’s
+``config.properties`` file, using one of the ``*user-mapping.pattern``
+properties from the table below that matches the authentication type of your
+configured authentication provider. For example, for an :doc:`LDAP
+</security/ldap>` authentication provider:
 
-Each authentication system has a separate property for the user mapping pattern to allow different mapping
-when multiple authentication systems are enabled:
+.. code-block:: text
+
+    http-server.authentication.password.user-mapping.pattern=(.*)(@.*)
+
+Remember that an :doc:`authentication type </security/authentication-types>`
+represents a category, such as ``PASSWORD``, ``OAUTH2``, ``KERBEROS``. More than
+one authentication method can have the same authentication type. For example,
+the Password file, LDAP, and Salesforce authentication methods all share the
+``PASSWORD`` authentication type.
+
+You can specify different user mapping patterns for different authentication
+types when multiple authentication methods are enabled:
 
 ===================================== ===============================================================
-Authentication                        Property
+Authentication type                   Property
 ===================================== ===============================================================
-Username and Password (file or LDAP)  ``http-server.authentication.password.user-mapping.pattern``
-Kerberos                              ``http-server.authentication.krb5.user-mapping.pattern``
+Password (file, LDAP, Salesforce)     ``http-server.authentication.password.user-mapping.pattern``
+OAuth2                                ``http-server.authentication.oauth2.user-mapping.pattern``
 Certificate                           ``http-server.authentication.certificate.user-mapping.pattern``
-Json Web Token                        ``http-server.authentication.jwt.user-mapping.pattern``
+JSON Web Token                        ``http-server.authentication.jwt.user-mapping.pattern``
+Kerberos                              ``http-server.authentication.krb5.user-mapping.pattern``
 Insecure                              ``http-server.authentication.insecure.user-mapping.pattern``
 ===================================== ===============================================================
 
+.. _pattern-file:
+
 File mapping rules
 ------------------
 
-The file mapping rules allow for more complex mappings from the authentication user.  These rules are
-loaded from a JSON file defined in a configuration property.  The mapping is based on the first matching
-rule, processed from top to bottom. If no rules match, authentication is denied.  Each rule is composed of the
+Use the **File mapping rules** method if your authentication provider’s way to
```

Comment:
Can we rewrite this to avoid the possessive  here?

Reply:
Done.